roles/2_setup-infra/tasks/subtasks/dependencies-debian.yml

```diff
@@ -11,6 +11,28 @@
     - sp_update_system is defined
     - sp_update_system | bool
 
+- name: Install specified kernel
+  apt:
+    pkg:
+      - "linux-image-{{ sp_kernel_version }}"
+    update_cache: yes
+  when:
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+- name: Install extra kernel modules
+  apt:
+    pkg:
+      - "linux-modules-extra-{{ sp_kernel_version }}"
+    update_cache: yes
+  when:
+    - ansible_distribution == "Ubuntu"
+    - sp_kernel_version is defined
+    - not ansible_kernel | regex_search("^" + sp_kernel_version)
+
+- name: Include tasks for setting default kernel
+  include_tasks: "kernel.yml"
```

Comment:
Please use a more elaborate naming scheme, for example ``boot_into_updated_kernel`` or something similar.